packages/twenty-server/src/core/user-workspace/user-workspace.service.ts

```diff
@@ -0,0 +1,81 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { TypeOrmQueryService } from '@ptc-org/nestjs-query-typeorm';
+import { Repository } from 'typeorm';
+
+import { UserWorkspace } from 'src/core/user-workspace/user-workspace.entity';
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { DataSourceService } from 'src/metadata/data-source/data-source.service';
+import { User } from 'src/core/user/user.entity';
+
+export class UserWorkspaceService extends TypeOrmQueryService<UserWorkspace> {
+  constructor(
+    @InjectRepository(UserWorkspace, 'core')
+    private readonly userWorkspaceRepository: Repository<UserWorkspace>,
+    private readonly dataSourceService: DataSourceService,
+    private readonly typeORMService: TypeORMService,
+  ) {
+    super(userWorkspaceRepository);
+  }
+
+  async create(userId: string, workspaceId: string): Promise<UserWorkspace> {
+    const userWorkspaces = await this.findUserWorkspaces(userId);
+
+    if (userWorkspaces.length > 0) {
+      const maxExistingPosition = Math.max(
+        ...userWorkspaces.map(({ position }) => position),
+      );
+      const newUserWorkspace = this.userWorkspaceRepository.create({
+        userId,
+        workspaceId,
+        position: maxExistingPosition + 1,
+      });
+
+      return this.userWorkspaceRepository.save(newUserWorkspace);
+    }
+
+    const userWorkspace = this.userWorkspaceRepository.create({
+      userId,
+      workspaceId,
+      position: 0,
+    });
+
+    return this.userWorkspaceRepository.save(userWorkspace);
+  }
+
+  async createWorkspaceMember(workspaceId: string, user: User) {
```

Comment:
Think you should remove the `createWorkspaceMember` from the user.service.ts then